core/src/queue.rs

```diff
@@ -82,6 +82,27 @@ pub struct Failure {
     pub err: Error,
 }
 
+/// Will remove transaction from the queue on drop.
+/// See [`Queue::remove_stale_transaction`] for details.
+pub struct TransactionGuard {
+    tx: AcceptedTransaction,
+    queue: Arc<Queue>,
+}
```

Comment:
```suggestion
pub struct TransactionGuard<'q> {
    tx: AcceptedTransaction,
    queue: &'q Queue,
}
```

wouldn't this have been possible?

Reply:
If we remove `Arc` from `TransactionGuard`, basically we will have this:
```rust
struct Queue {}

struct Sumeragi<'q> {
    queue: Arc<Queue>,
    transaction_cache: Vec<&'q Queue>,
}

fn run(mut sumeragi: Sumeragi) {
    sumeragi.transaction_cache.push(&sumeragi.queue);
}
```

I don't see how this can pass borrow checker